addons/sale_margin/models/sale_order.py

```diff
@@ -19,6 +19,8 @@ class SaleOrderLine(models.Model):
     @api.depends('product_id', 'company_id', 'currency_id', 'product_uom')
     def _compute_purchase_price(self):
         for line in self:
+            if line.purchase_price:
+                continue
```

Comment:
Nope, we want the recomputation when the product/company/currency/uom is modified...
If there is a specific timesheet case, it should avoid calling super for the line where the recomputation shouldn't happen, but not disable the computation.
With your change, the compute dependencies simply become useless :confused: (except for the case when the computed price was 0).

Reply:
I'll work on a better solution that does not skip the computation but keep the purchase_price is set.

I will also create a test.

Thanks for the review!